pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SignatureDeclareThrowsRuntimeExceptionRule.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule.design;
+
+import java.util.Collections;
+import java.util.List;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTName;
+import net.sourceforge.pmd.lang.java.ast.ASTNameList;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
+import net.sourceforge.pmd.lang.java.typeresolution.TypeHelper;
+
+/**
+ * A method/constructor should not explicitly declare java.lang.RuntimeException or 
+ * it's subclasses in throws clause of it's signature, since it is advised to avoid 
+ * declaring unchecked exceptions in method signature.
+ *
+ * @author <a href="prakashbp2020@gmail.com">Bhanu Prakash Pamidi</a>
+ */
+public class SignatureDeclareThrowsRuntimeExceptionRule extends AbstractJavaRule {
+
```

Comment:
Since this rule only visits two AST nodes, we can make use of the "RuleChain" mechanism. This means, that not the whole AST tree is traversed and visited, but only specific AST nodes. This speeds up the rule execution time.

You declare the nodes, you want to have considered, in the default constructor of your rule:

```java
    public SignatureDeclareThrowsRuntimeExceptionRule() {
        addRuleChainVisit(ASTMethodDeclaration.class);
        addRuleChainVisit(ASTConstructorDeclaration.class);
    }
```

Then, when implementing the visit method, don't call super - since we don't want to traverse the AST anymore:

```java
    @Override
    public Object visit(ASTMethodDeclaration methodDeclaration, Object o) {
        checkExceptions(methodDeclaration, o);
        return o;
    }
```